src/ansys/grantami/bomanalytics/_item_results.py

```diff
@@ -732,9 +745,10 @@ def __init__(
 
         for legislation in legislations:
             new_substances = [
-                self._create_impacted_substance(substance) for substance in legislation.impacted_substances
+                self._create_impacted_substance(substance)
+                for substance in _cast_unset_union_to_any(legislation.impacted_substances)
```

Comment:
Github won't let me expand the code before so I can't really see what's what. I'll try to check locally.

Is the casting necessary here? I can't see how `legislation.impacted_substances` could be None without this block raising an error.
I suppose the question is, was the casting applied based on test failures, mypy errors, or everywhere?

Reply:
Yes, that's correct. My rationale being that in previous releases it was either a list or `None`, and we previously didn't check it wasn't `None`, so it seemed safe to change it to what we have now.

That's my philosophy behind all these changes:

* If we previously didn't check that a thing was None, then pass it through the generic function that raises if it's Unset. At least now it will fail with an explicit error message, instead of a `None type is not indexable` or whatever we'd get
* If we previously did check that a thing was None, then pass it through the generic function that converts Unset to None. I've created an openapi-client-template PR that includes this conversion in the models themselves, which would make this generic function unnecessary

Reply:
https://github.com/ansys/openapi-client-template/pull/168

Reply:
The first one, basically the vast majority of fields are not required and not nullable, so they end up with this Unset union type. It would be nice to get it fixed in the server definition - I will add a ticket in Jira to this effect.

Reply:
And no, Unset really only makes sense for models used for patch requests. The problem is, we don't distinguish between what a model is used for (request/response/verb), and so we're kind of stuck with it. The additional parallel models is intended to be a way of reconciling that.

Reply:
I'm not sure I understand the point of `_raise_if_unset`: if we were previously not checking if something was None (because the API always includes even though it does not document it as such), why do we now need to check that it's not Unset?
Especially since, if it is, we'll now raise a cryptic ValueError

Reply:
I thought it was better to explicitly raise a ValueError indicating that the value received from the server was wrong, as opposed to a TypeError because whatever we happened to try to do to the incorrect value wasn't possible based on its type,

Although I realize now looking at the code that I didn't actually put a message in with the ValueError. This should probably be something like:

`raise ValueError("Provided value cannot be Unset")`

Maybe passing the name of the field as an extra argument, so we can raise with a more informative message?

Reply:
I should also add, the reason we raise is to satisfy mypy. The alternative would be to cast to `T`, which I think will work.